pkg/resources/resource_monitor.go

```diff
@@ -2,504 +2,450 @@ package resources
-		accountOpts := sdk.AlterAccountOptions{
-			Set: &sdk.AccountSet{
-				ResourceMonitor: sdk.NewAccountObjectIdentifier("NULL"),
-			},
-		}
-		if err := client.Accounts.Alter(ctx, &accountOpts); err != nil {
-			return fmt.Errorf("error unsetting resource monitor %v on account err = %w", objectIdentifier.Name(), err)
-		}
-	}
+	if d.HasChanges("notify_triggers", "suspend_trigger", "suspend_immediate_trigger") {
+		triggers := make([]sdk.TriggerDefinition, 0)
 
-	// Remove from all old warehouses
-	if d.HasChange("warehouses") {
-		oldV, v := d.GetChange("warehouses")
-		res := ADiffB(oldV.(*schema.Set).List(), v.(*schema.Set).List())
-		for _, w := range res {
-			warehouseOpts := sdk.AlterWarehouseOptions{
-				Unset: &sdk.WarehouseUnset{
-					ResourceMonitor: sdk.Bool(true),
-				},
-			}
-			warehouseId := sdk.NewAccountObjectIdentifier(w)
-			if err := client.Warehouses.Alter(ctx, warehouseId, &warehouseOpts); err != nil {
-				return fmt.Errorf("error unsetting resource monitor %v on warehouse %v err = %w", name, warehouseId.Name(), err)
+		if notifyTriggers, ok := d.GetOk("notify_triggers"); ok {
+			for _, triggerThreshold := range notifyTriggers.(*schema.Set).List() {
+				triggers = append(triggers, sdk.TriggerDefinition{
+					Threshold:     triggerThreshold.(int),
+					TriggerAction: sdk.TriggerActionNotify,
+				})
 			}
 		}
-	}
 
-	// Add to account
-	if d.HasChange("set_for_account") && d.Get("set_for_account").(bool) {
-		accountOpts := sdk.AlterAccountOptions{
-			Set: &sdk.AccountSet{
-				ResourceMonitor: objectIdentifier,
-			},
+		if suspendTriggerThreshold, ok := d.GetOk("suspend_trigger"); ok {
+			triggers = append(triggers, sdk.TriggerDefinition{
+				Threshold:     suspendTriggerThreshold.(int),
+				TriggerAction: sdk.TriggerActionSuspend,
+			})
 		}
-		if err := client.Accounts.Alter(ctx, &accountOpts); err != nil {
-			return fmt.Errorf("error setting resource monitor %v on account err = %w", name, err)
+
+		if suspendImmediateTriggerThreshold, ok := d.GetOk("suspend_immediate_trigger"); ok {
+			triggers = append(triggers, sdk.TriggerDefinition{
+				Threshold:     suspendImmediateTriggerThreshold.(int),
+				TriggerAction: sdk.TriggerActionSuspendImmediate,
+			})
 		}
-	}
 
-	// Add to all new warehouses
-	if d.HasChange("warehouses") {
-		oldV, v := d.GetChange("warehouses")
-		res := ADiffB(v.(*schema.Set).List(), oldV.(*schema.Set).List())
-		for _, w := range res {
-			warehouseOpts := sdk.AlterWarehouseOptions{
-				Set: &sdk.WarehouseSet{
-					ResourceMonitor: objectIdentifier,
+		if len(triggers) > 0 {
+			opts.Triggers = triggers
+		} else {
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  "Failed to update resource monitor.",
+					Detail:   "Due to Snowflake limitations triggers cannot be completely removed form resource monitor after having at least 1 trigger. The only way it to re-create resource monitor without any triggers specified.",
```

Comment:
Add this to the docs as well

Reply:
I touch the documentation in the next pr, I'll address this comment there.